01.md

```diff
@@ -35,6 +35,10 @@ Once `LNURL` is decoded:
 
 Neither status codes or any HTTP Header has any meaning. Servers may use whatever they want. Clients should ignore them (and be careful when using libraries that treat responses differently based on headers and status codes) and just parse the response body as JSON, then interpret it accordingly.
 
+## CORS Browser Compatibility
+
+Application endpoints or Reverse Proxy directives that are accessed when using the LNURL must respond to `OPTIONS` requests on that endpoint. In their replies to `GET` and `OPTIONS`, they must set the header `Access-Control-Allow-Origin: *` for LNURL's to work with browser-based wallets, services and other webapps. If they set the `Access-Control-Allow-Methods` header, they must set in such a way that `GET` requests are permitted.
```

Comment:
```suggestion
Application endpoints that are accessed when using the LNURL must respond to `OPTIONS` requests on that endpoint. In their replies to `GET` and `OPTIONS`, they must set the header `Access-Control-Allow-Origin: *` for LNURL's to work with browser-based wallets, services and other webapps. If they set the `Access-Control-Allow-Methods` header, they must set in such a way that `GET` requests are permitted.
```

I still disagree with the reference to reverse proxies. A reverse proxy is something completely separate and controlled by the user or platform, not by the application developer. 

A reverse proxy is not always part of the application stack, and not under the control of an app developer. For example, BTCPay, which supports LNUrl, can be run on StartOS, Umbrel, ... which may already include a reverse proxy.

Requiring this for apps themselves means apps are correct without users externally modifying something about the app's responses.

If we impose this rule on reverse proxies, a lot of people are going to set `Access-Control-Allow-Origin: *` for all routes, which will make apps insecure.

Clearly putting the responsibility on the app developer is much better imo.

Reply:
That's a very niche subset of reverse proxies and afaik they function more as gateways than reverse proxies, an umbrel user exposing services to the internet would need to reconfigure the reverse proxy anyway for things like SSL and so on... I also explicitly mentioned the directive for the endpoint, of which there would be many in an box like that for various apps

Reply:
Still, these proxies are controlled by the user and not the application developer, who is responsible for implementing the spec.

Reply:
By exposing an umbrel etc to the internet the user is now a service provider, the box is the application including the customized reverse proxy configuration

It's services that need these fixes, not users.

Reply:
> By exposing an umbrel etc to the internet the user is now a service provider, the box is the application including the customized reverse proxy configuration
> 
> It's services that need these fixes, not users.

Every other part of the spec is implemented by app developers, not users. All of [these services](https://github.com/lnurl/luds?tab=readme-ov-file#self-hosted) have implemented LNURL. No user configured the LNURL handling in their reverse proxy or has to read the spec. Application & library developers have to know them, however. 

The application **can** do this, and the reverse proxy is not needed for such functionality, so it should be part of the app.

Saying "our app may support LNURL" is incredibly bad as a developer and complicates setup for users unnecessarily.

Reply:
If we're being precise then those are apps, not services

I agree that apps should implement this, but the reality is that many apps are unlikely to be updated for this and are harder for the service provider to update themselves vs. their reverse proxy configuration

Reply:
> If we're being precise then those are apps, not services
> 
> I agree that apps should implement this, but the reality is that many apps are unlikely to be updated for this and are harder for the service provider to update themselves vs. their reverse proxy configuration

I think the spec should talk about apps, and then service providers can fix invalid apps if they choose so, but that should be outside of the spec.

Reply:
Apps are geared towards services, services which are run by operators that may be trying to figure out why teir app isn't working in certian cases- including searching on the CORS error from their browser

A reference to app directives gives both sides the information they need, and doesn't imply app developers shouldn't do it directly

Reply:
But it may lead app developers to put the responsibility onto users. Not mentioning reverse proxies makes it clear app developers must do it.

Reply:
The responsibility is already on operators of broken services to fix them, a bad app developer or unmaintained app will deviate no matter whats added to the spec and so verbiage is warranted for both configurations

It's also fair for an app to simply document the external requirement just as it might SSL